Userland/Applications/PixelPaint/Guide.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2021, Tobias Christiansen <tobi@tobyase.de>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/NonnullRefPtr.h>
+#include <AK/RefCounted.h>
+
+namespace PixelPaint {
+
+class Guide : public RefCounted<Guide> {
+public:
+    enum class Orientation {
+        Vertical,
+        Horizontal,
+    };
+
+    static NonnullRefPtr<Guide> create(Orientation orientation, float offset)
+    {
+        return adopt_ref(*new Guide(orientation, offset));
```

Comment:
```suggestion
        return create<Guide>(orientation, offset);
```

Reply:
If I understand correctly this isn't completely wrong, just the more Serenity way?
We do get a NonnullRefPtr in both cases after all...

Reply:
Yes, the naked `new` is just ugly :)